lib/Interpreter/Transaction.cpp

```diff
@@ -64,12 +64,23 @@ namespace cling {
   }
 
   NamedDecl* Transaction::containsNamedDecl(llvm::StringRef name) const {
-    for (auto I = decls_begin(), E = decls_end(); I < E; ++I)
+    for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
-        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI))
+        if ( DI->getKind() == Decl::LinkageSpec ) {
```

Comment:
Could you remove the spaces after ( and ). Also could you use isa<LinkageSpecDecl>(DI) instead?